source/isaaclab_mimic/isaaclab_mimic/motion_planners/curobo/curobo_planner_config.py

```diff
@@ -0,0 +1,459 @@
+                collision_spheres_file="spheres/my_robot_spheres.yml",  # Path relative to curobo (can override with custom spheres file)
+
+                # Grasp detection threshold
+                grasp_gripper_open_val=0.05,
+
+                # Motion planning parameters (tune for your robot)
+                approach_distance=0.05,  # Distance to approach before grasping
+                retreat_distance=0.05,   # Distance to retreat after grasping
+                time_dilation_factor=0.5,  # Speed factor (0.5 = half speed)
+
+                # Visualization options
+                visualize_spheres=False,
+                visualize_plan=False,
+                debug_planner=False,
+            )
+
+    2. Task-Specific Configurations
+    --------------------------------
+    For task-specific variants, create methods that modify the base config:
+
+    .. code-block:: python
+
+        @classmethod
+        def my_robot_pick_place_config(cls) -> "CuroboPlannerConfig":
+            config = cls.my_robot_config()  # Start from base config
+
+            # Override for pick-and-place tasks
+            config.approach_distance = 0.08
+            config.retreat_distance = 0.10
+            config.enable_finetune_trajopt = True
+            config.collision_activation_distance = 0.02
+
+            # Custom world configuration if needed
+            config.get_world_config = lambda: config._get_world_config_with_table_adjustment()
+
+            return config
+
+    3. Register in from_task_name()
+    --------------------------------
+    Add your robot detection logic to the from_task_name method:
+
+    .. code-block:: python
+
+        @classmethod
+        def from_task_name(cls, task_name: str) -> "CuroboPlannerConfig":
+            task_lower = task_name.lower()
+
+            # Add your robot detection
+            if "my-robot" in task_lower:
+                if "pick-place" in task_lower:
+                    return cls.my_robot_pick_place_config()
+                else:
+                    return cls.my_robot_config()
+
+            # ... existing robot checks ...
+
+    Important Notes
+    ---------------
+    - The _create_temp_robot_yaml() helper creates a temporary YAML with your custom URDF
+    - If using Nucleus assets, retrieve_file_path() downloads them to a local temp directory
+    - The base robot YAML (e.g., "my_robot.yml") should exist in cuRobo's robot configs
+
+    Best Practices
+    --------------
+    1. Start with conservative parameters (slow speed, large distances)
+    2. Test with visualization enabled (visualize_plan=True) for debugging
+    3. Tune collision_activation_distance based on controller precision to follow collision-free motion
+    4. Adjust sphere counts in extra_collision_spheres for attached objects
+    5. Use debug_planner=True when developing new configurations
+    """
+
+    @classmethod
+    def franka_config(cls) -> "CuroboPlannerConfig":
```

Comment:
Personally, instead of having these factory methods which will end up blowing up as more robot support is added, I think it is better to have separate config objects defined, similar to how we do for assets and other pieces of the codebase.